cookbook/doctrine/registration_form.rst

```diff
@@ -138,230 +170,143 @@ Next, create the form for the ``User`` model::
         }
     }
 
-There are just two fields: ``email`` and ``plainPassword`` (repeated to confirm
-the entered password). The ``data_class`` option tells the form the name of the
-underlying data class (i.e. your ``User`` entity).
+There are just three fields: ``email``, ``username`` and ``plainPassword``
+(repeated to confirm the entered password).
 
 .. tip::
 
     To explore more things about the Form component, read :doc:`/book/forms`.
 
-Embedding the User Form into a Registration Form
-------------------------------------------------
-
-The form that you'll use for the registration page is not the same as the
-form used to simply modify the ``User`` (i.e. ``UserType``). The registration
-form will contain further fields like "accept the terms", whose value won't
-be stored in the database.
+Handling the Form Submission
+----------------------------
 
-Start by creating a simple class which represents the "registration"::
+Next, you need a controller to handle the form. Start by creating a simple
+controller for displaying the registration form::
 
-    // src/Acme/AccountBundle/Form/Model/Registration.php
-    namespace Acme\AccountBundle\Form\Model;
+    // src/AppBundle/Controller/AccountController.php
+    namespace AppBundle\Controller;
 
-    use Symfony\Component\Validator\Constraints as Assert;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 
-    use Acme\AccountBundle\Entity\User;
+    use AppBundle\Form\UserType;
+    use AppBundle\Entity\User;
+    use Symfony\Component\HttpFoundation\Request;
+    use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 
-    class Registration
+    class RegistrationController extends Controller
     {
         /**
-         * @Assert\Type(type="Acme\AccountBundle\Entity\User")
-         * @Assert\Valid()
-         */
-        protected $user;
-
-        /**
-         * @Assert\NotBlank()
-         * @Assert\True()
+         * @Route("/register", name="user_registration")
          */
-        protected $termsAccepted;
-
-        public function setUser(User $user)
+        public function registerAction(Request $request)
         {
-            $this->user = $user;
-        }
-
-        public function getUser()
-        {
-            return $this->user;
-        }
-
-        public function getTermsAccepted()
-        {
-            return $this->termsAccepted;
-        }
+            // 1) build the form
+            $user = new User();
+            $form = $this->createForm(new UserType(), $user);
 
-        public function setTermsAccepted($termsAccepted)
-        {
-            $this->termsAccepted = (Boolean) $termsAccepted;
-        }
-    }
+            // 2) handle the submit (will only happen on POST)
+            $form->handleRequest($request);
+            if ($form->isValid()) {
```

Comment:
should be `if ($form->isSubmitted() && $form->isValid())` to comply with the best practices